Firestore/src/Aggregate.php

```diff
@@ -106,18 +103,31 @@ public static function sum($field)
      * ```
      * $avg = Aggregate::avg('field_to_aggregate_upon');
      * ```
+     *
+     * Average of empty valid data set return `null`.
+     * Average of numbers exceeding max float value returns `INF`.
+     * Average of data which contains `NaN` returns `NaN`.
+     *
      * @param string $field The relative path of the field to aggregate upon.
      * @return Aggregate
      */
     public static function avg($field)
     {
-        $avg = new Aggregate(self::TYPE_AVG);
-        $avg->props[$avg->aggregationType] = [
-            'field' => [
-                'fieldPath' => $field
-            ]
-        ];
-        return $avg;
+        return self::createAggregate(self::TYPE_AVG, $field);
+    }
+
+    private static function createAggregate(string $type, string $field = '')
```

Comment:
Can we use null as default for `$field`?

Reply:
Thanks a lot for noticing this!

Didn't know `empty("0")` return `true`.